File: src/main/kotlin/by/mksn/kwitapi/repository/UserRepository.java
package by.mksn.kwitapi.repository;

import by.mksn.kwitapi.entity.User;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Component;

@Component
public interface UserRepository extends CrudRepository<User, Long> {

    User findByUsername(String username);

}

File: src/main/kotlin/by/mksn/kwitapi/repository/UserRepository.kt
package by.mksn.kwitapi.repository

import by.mksn.kwitapi.entity.User
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : PagingAndSortingRepository<User, Long> {
    fun findByEmail(email: String): User?
}